src/contracts/middleware/BLSRegistryCoordinatorWithIndices.sol

```diff
@@ -0,0 +1,352 @@
+        bytes32 registeringOperatorId = _operators[msg.sender].operatorId;
+
+        // kick the operators
+        for (uint256 i = 0; i < quorumNumbers.length; i++) {
+            // check that the quorum has reached the max operator count
+            uint8 quorumNumber = uint8(quorumNumbers[i]);
+            OperatorSetParam memory operatorSetParam = _quorumOperatorSetParams[quorumNumber];
+            {
+                uint32 numOperatorsForQuorum = indexRegistry.totalOperatorsForQuorum(quorumNumber);
+                require(
+                    numOperatorsForQuorum == operatorSetParam.maxOperatorCount + 1,
+                    "BLSIndexRegistryCoordinator.registerOperatorWithCoordinator: quorum has not reached max operator count"
+                );
+
+                // get the total stake for the quorum
+                uint96 totalStakeForQuorum = stakeRegistry.getCurrentTotalStakeForQuorum(quorumNumber);
+                bytes32 operatorToKickId = _operators[operatorKickParams[i].operator].operatorId;
+                uint96 operatorToKickStake = stakeRegistry.getCurrentOperatorStakeForQuorum(operatorToKickId, quorumNumber);
+                uint96 registeringOperatorStake = stakeRegistry.getCurrentOperatorStakeForQuorum(registeringOperatorId, quorumNumber);
+
+                // check the registering operator has more than the kick BIPs of the operator to kick's stake
+                require(
+                    registeringOperatorStake > operatorToKickStake * operatorSetParam.kickBIPsOfOperatorStake / BIPS_DENOMINATOR,
+                    "BLSIndexRegistryCoordinator.registerOperatorWithCoordinator: registering operator has less than kickBIPsOfOperatorStake"
+                );
+                
+                // check that the operator to kick has less than the kick BIPs of the average stake
+                require(
+                    operatorToKickStake < (totalStakeForQuorum * operatorSetParam.kickBIPsOfAverageStake / BIPS_DENOMINATOR) / numOperatorsForQuorum,
+                    "BLSIndexRegistryCoordinator.registerOperatorWithCoordinator: operator to kick has more than kickBIPsOfAverageStake"
+                );
+                // check the that the operator to kick has less than the kick BIPs of the total stake
+                require(
+                    operatorToKickStake < totalStakeForQuorum * operatorSetParam.kickBIPsOfTotalStake / BIPS_DENOMINATOR,
+                    "BLSIndexRegistryCoordinator.registerOperatorWithCoordinator: operator to kick has more than kickBIPSOfTotalStake"
+                );
+            }
+            
+            // kick the operator
+            _deregisterOperatorWithCoordinator(
+                operatorKickParams[i].operator, 
+                quorumNumbers[i:i+1], 
+                operatorKickParams[i].pubkey, 
+                operatorKickParams[i].operatorIdsToSwap, 
+                operatorKickParams[i].globalOperatorListIndex
+            );
```

Comment:
thinking more about late reverts, this seems like it could get pretty bad with race conditions :/

Reply:
I guess the idea is primarily to set the OperatorSetParameters appropriately so there isn't constant churn

Reply:
mm OK. I'm fine leaving as-is for now, although I like the thought.